core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -416,6 +435,21 @@ object Uri extends UriPlatform {
         }
     }
 
+    trait SegmentEncoder[A] {
+      def toSegment(a: A): Segment
+    }
+    object SegmentEncoder {
+      implicit def apply[A](implicit segmentEncoder: SegmentEncoder[A]): SegmentEncoder[A] =
+        segmentEncoder
+
+      def instance[A](f: A => Segment): SegmentEncoder[A] = new SegmentEncoder[A] {
+        override def toSegment(a: A): Segment = f(a)
+      }
+
```

Comment:
Good suggestion.

What about something like 
```
object AlwaysFromShow {
    implicit def fromShow[A](implicit A: Show[A]): SegmentEncoder[A] = instance(a => Segmant(Show.show(a)))
}
```

So if someone wants to just always use a show instance there is an import they can use?

Reply:
I'd not do that personally. 

So many `Show` instances don't have the correct form for a URI, e.g. anything that Kittens generates will be like,

```scala
final case class Foo(bar: String)

Foo("baz").show === "Foo(bar = baz)"
```

e.g. with the named arguments in the `Show`.

Plus, if someone wants this behavior, it is really easy for them to add an orphan instance in their local scope.

Reply:
It's a good thought but IMO an import like that could easily be more trouble than helpful. A couple reasons:
1. You'd have to include that import everywhere you want a `SegmentEncoder` for a particular `A`. Whereas if you add an implicit `SegmentEncoder[A]` defined via `fromShow` to the companion object for `A` it will always be in scope without additional imports.
2. This import may cause ambiguous implicit conflicts with other `SegmentEncoder`s or even worse silently override them and replace with something invalid.

Reply:
👍